### ISSUE TITLE
[JSON API] ED doesn't consider `links` in the compound documents

### ISSUE THREAD

**Issue by lancedikson** [emberjs/data#5062]
It seems that ED doesn't recognize `links` when `data` exists as well in the response from the server. The example of a buggy response from server: 

```js
{
  "data": {
  	...
  },
  "included": [
    {
      "type": "blog-posts",
      "id": "1",
      "attributes": {
        "title": "Lorem"
      },
      "relationships": {
        "comments": {
          "links": {
            "related": "/api/blog-posts/1/comments"
          },
          "data": [
            {
              "type": "comments",
              "id": "1"
            },
            {
              "type": "comments",
              "id": "2"
            },
            {
              "type": "comments",
              "id": "3"
            }
          ]
        }
      }
    }
  ]
}
```

In that case, ED is trying to send requests to `/api/comments/1` ignoring the `links.related` attribute. But, seems that it should recognize that kind of links according to [the specification](http://jsonapi.org/format/#document-compound-documents). Also, I created [a discussion on the JSON API forum](http://discuss.jsonapi.org/t/could-links-and-data-persist-in-compound-documents-relations/1023).

**Comment by stefanpenner** [comment]
This seems quite close to the test scenario provided in https://github.com/emberjs/data/pull/5147 can you confirm that is the same scenario? If so, https://github.com/emberjs/data/pull/5147 fixes it. If not, if you can provide a failing test, I can investigate.

**Comment by stefanpenner** [comment]
This seems quite close to the test scenario provided in https://github.com/emberjs/data/pull/5147 can you confirm that is the same scenario? If so, https://github.com/emberjs/data/pull/5147 fixes it. If not, if you can provide a failing test, I can investigate.

**Comment by andrewfan** [comment]
Hey, @stefanpenner @lancedikson, I think `links.related` still does not work, I did not check the whole flow of serialization to understand if `links` from response are properly stored somewhere, but I checked https://github.com/emberjs/data/blob/master/addon/-private/adapters/build-url-mixin.js#L55. `buildURL` method is called for every API operation and it does not have any logic for handling `links`. @stefanpenner do you know if this feature will be supported in the near future?

**Comment by runspired** [comment]
This is resolved by #5410 🎉
